Revocation and Status Assertions flows reviewed

SaraConsoliACN opened the pull request:
## Title
Revocation and Status Attestation flows reviewed
## Content
Content review based on the new version of the specifications defined in https://github.com/peppelinux/draft-demarco-oauth-status-attestations

resolve #265

SaraConsoliACN committed:
Status assertion first draft

SaraConsoliACN committed:
status assertion response completed

SaraConsoliACN committed:
Editorial change

SaraConsoliACN committed:
Revocation flow completed

SaraConsoliACN committed:
Editorial review

SaraConsoliACN committed:
Specification tables JSON objects added

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -212,29 +227,59 @@ The following HTTP Status Codes MUST be supported:
+    "typ": "revocation-assertion-error+jwt",
+    "kid": "Issuer-JWK-KID"
+  }
+	.
```
```suggestion
.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -212,29 +227,59 @@ The following HTTP Status Codes MUST be supported:
+    "jti": "6f204f7e-e453-4dfd-814e-9d155319408c"
+    "credential_hash": $CREDENTIAL-HASH,
+    "credential_hash_alg": "sha-256",
+    "msg": "Revoked"
```
msg?

peppelinux replied:
@SaraConsoliACN please, as discussed during our editor call, provide another parameter name with a good semantic

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -244,109 +289,142 @@ The following Error Codes MUST be supported:
+    "typ": "revocation-assertion-error+jwt",
+    "kid": "Issuer-JWK-KID"
+  }
+	.
```
```suggestion
.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -244,109 +289,142 @@ The following Error Codes MUST be supported:
     POST /status HTTP/1.1
-    Host: pid-provider.example.org
-    Content-Type: application/x-www-form-urlencoded
+	Host: issuer.example.org
+	Content-Type: application/json
```
```suggestion
    Host: issuer.example.org
    Content-Type: application/json
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -503,55 +606,137 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+    * - **jti**
+      - Unique identifier for the JWT.
+      - `[RFC7519, Section 4.1.7] <https://www.iana.org/go/rfc7800>`_.
+    * - **msg**
+      - Message returned from the Credential Issuer after revocation. It MUST be set with the value "OK".
+      - This specification.
```
????

peppelinux replied:
this msg is not defined in the status assertion draft.

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -503,55 +606,137 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+    - Algorithm used to verify the cryptographic signature of the Revocation Assertion. Revocation Assertion that do not need to be signed SHOULD set the `alg` value to `none` in according with `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
+    - `[OIDC4VCI. Draft 13] <https://openid.bitbucket.io/connect/openid-4-verifiable-credential-issuance-1_0.html>`_, [:rfc:`7515`], [:rfc:`7517`].
+  * -  **typ** 
+    - It MUST be set to `revocation-assertion-respone+jwt` when JWT format is used. It MUST be set to `revocation-assertion-response+cwt` when CWT format is used.
```
```suggestion
    - It MUST be set to `revocation-assertion-response+jwt` when JWT format is used. It MUST be set to `revocation-assertion-response+cwt` when CWT format is used.
```

SaraConsoliACN committed:
Accepted review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Accepted review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Accepted review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Accepted review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -131,26 +132,30 @@ A Wallet Instance MUST request the revocation of a Digital Credential as defined
     {
       "alg": "ES256",
-      "typ": "status-attestation-request+jwt",
+      "typ": "revocation-assertion-request+jwt",
```
```suggestion
      "typ": "credential-revocation-request+jwt",
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -164,23 +169,26 @@ Below, is given a non-normative example of a Credential PoP with decoded JWT hea
 .. code::
 
     .. code-block:: http
+		HTTP/1.1 200 Created
```
```suggestion
		HTTP/1.1 200 Ok
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -164,23 +169,26 @@ Below, is given a non-normative example of a Credential PoP with decoded JWT hea
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The requests to the *Issuer Revocation endpoint* MUST be HTTP with method POST, using the mandatory parameters listed below within the HTTP request message body. These MUST be encoded in ``application/x-www-form-urlencoded`` format.
+The requests to the *Credential Issuer Revocation endpoint* MUST be HTTP with method POST, using the mandatory parameters listed below within the HTTP request message body. These MUST be encoded in ``application/JSON`` format.
```
```suggestion
The requests to the *Credential Issuer Revocation endpoint* MUST be HTTP with method POST, using the mandatory parameters listed below within the HTTP request message body. These MUST be encoded in ``application/json`` format.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -190,18 +198,25 @@ The requests to the *Issuer Revocation endpoint* MUST be HTTP with method POST,
-    * - **credential_pop**
-      - It MUST contain a JWT proof of possession of the cryptographic key the Credential to be revoked shall be bound to. See Section :ref:`Credential Proof of Possession <sec_revocation_credential_pop>` for more details. 
-      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_
+    * - **revocation_assertion_requests**
```
this parameter name must be changed

the scope of the request is to request the revocation of credentials or wallet instance attestations

SaraConsoliACN replied:
the parameter name is changed with revocation_requests

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -190,18 +198,25 @@ The requests to the *Issuer Revocation endpoint* MUST be HTTP with method POST,
-      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_
+    * - **revocation_assertion_requests**
+      - It MUST be an array of strings, where each represents a Revocation Assertion Request object. Each element MUST contain a signed JWT/CWT as a cryptographic proof of possession to which the Digital Credential to be revoked shall be bound. See Section :ref:`Credential Proof of Possession <sec_revocation_credential_pop>` for more details. 
+      - `[OAuth Status Assertion draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_
```
```suggestion
      - `[OAuth Status Assertion draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-assertions/01/>`_
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -212,29 +227,59 @@ The following HTTP Status Codes MUST be supported:
+    * - **Claim**
+      - **Description**
+      - **Reference**
+    * - **revocation_assertion_responses**
```
here the same as my previous comment, this parameter name must be changed

SaraConsoliACN replied:
As mentioned in the comment https://github.com/italia/eudi-wallet-it-docs/pull/308#discussion_r1658365625, the parameter name in changed.

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

peppelinux commented:
this PR must cover the wallet instance revocation request use case and the pid provider revocation request use case

SaraConsoliACN committed:
Alignment with the "draft-demarco-oauth-status-assertions" specifications

SaraConsoliACN committed:
white spaces fixed

cmarco0 committed:
update status code 404 not fount description in status attestation http response section

cmarco0 committed:
Add Credential PoP extra claims

SaraConsoliACN committed:
Merge branch 'SaraConsoliACN-statusAssertion' into cmarco0-ExtraClaimnew

SaraConsoliACN committed:
Merge pull request #2 from cmarco0/cmarco0-ExtraClaimnew

Credential PoP extra claims

SaraConsoliACN committed:
Merge branch 'SaraConsoliACN-statusAssertion' into cmarco0-404statuscodenew

SaraConsoliACN committed:
Merge pull request #1 from cmarco0/cmarco0-404statuscodenew

update status code '404 not found' description is status attestation HTTP response section

SaraConsoliACN committed:
Editorial review

SaraConsoliACN committed:
code fixed

SaraConsoliACN committed:
tab spaces fixed

SaraConsoliACN committed:
Editorial review

SaraConsoliACN committed:
Editorial changes

SaraConsoliACN committed:
Duplicate explicit target name and table structure fixed

SaraConsoliACN committed:
code fixed

SaraConsoliACN committed:
duplicate table name fixed

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -11,26 +11,26 @@ For these reasons a robust mechanism for managing the life-cycle and the revocat
 Furthermore, it provides the technical details that the Verifiers MUST implement to verify, in a secure and reliable manner, the validity of a Digital Credential during the presentation phase.
 
-The verification of the validity of a Digital Credential is based on the `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_ specification. 
+The verification of the validity of a Digital Credential is based on the `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_ specification.
```
```suggestion
The verification of the validity of a Digital Credential is based on the `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-oauth-status-assertions>`_ specification.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -40,18 +40,20 @@ Operational Requirements
+- MUST have a validity period not greater than 24 hours;
+- MUST provide the proof about the non-revocation of the Digital Credential which is related to and MUST be validated using the cryptographic signature of the Credential Issuer;
+- MUST NOT reveal any information about the Relying Party, the User's device or the User's data contained in the Digital Credential the assertion is related to;
+- MUST be non-repudiable even beyond its expiration time and even in the case of cryptographic keys rotation.
```
don't redefine the same requirements but point to the section defined in the status assertions draft

m-basili replied:
```suggestion
- MUST have a validity period not greater than 24 hours;
- MUST NOT reveal any information about the Relying Party, the User's device or the User's data contained in the Digital Credential the assertion is related to;
- MUST be non-repudiable even beyond its expiration time and even in the case of cryptographic keys rotation.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -97,14 +99,14 @@ Credential Revocation Flows can start under different scenarios, such as:
-    - The Authentic Sources that for any administrative reasons update one or more attributes of a User, shall inform the Issuer of related Credentials. 
-    - The Issuers, for technical security reasons (e.g. in the case of compromised cryptographic keys, death of the User, etc.), can decide to revoke the Credentials.
+    - The Authentic Sources that for any administrative reasons update one or more attributes of a User, shall inform the Credential Issuer of related Credentials. 
+    - The Credential Issuers, for technical security reasons (e.g. in the case of compromised cryptographic keys, death of the User, etc.), can decide to revoke the Credentials.
```
dead of the user is not a technical reason

can -> SHOULD

SaraConsoliACN replied:
The commit https://github.com/italia/eudi-wallet-it-docs/pull/308/commits/98136f003cb59e60bb9917ac9b8fd9c649af7334 aims to solve this conversation.

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -97,14 +99,14 @@ Credential Revocation Flows can start under different scenarios, such as:
     - The Law-Enforcing Authorities, for the fulfillment of their functions and any other judicial reasons, may request the Authentic Source to revoke entitlements, licenses, certificates, identification documents, etc., which in turn leads to the revocation of any linked Credentials.
-    - The Authentic Sources that for any administrative reasons update one or more attributes of a User, shall inform the Issuer of related Credentials. 
-    - The Issuers, for technical security reasons (e.g. in the case of compromised cryptographic keys, death of the User, etc.), can decide to revoke the Credentials.
+    - The Authentic Sources that for any administrative reasons update one or more attributes of a User, shall inform the Credential Issuer of related Credentials. 
```
```suggestion
    - The Authentic Sources that for any update of one or more User attributes, shall inform the Credential Issuer that has previously requested those data for the issuance of a Credential about that User. 
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -97,14 +99,14 @@ Credential Revocation Flows can start under different scenarios, such as:
 
     - The **Revocation flows**: these flows describe how an Entity requests for a Digital Credential revocation. 
-    - The **Status Attestation flows**: these flows define the technical protocols for requesting and obtaining a Status Attestation and how the Wallet Instance will provide it to a Verifier as a proof of validity of a corresponding Digital Credential.
+    - The **Status Assertion flows**: these flows define the technical protocols for requesting and obtaining a Status Assertion and how the Wallet Instance will provide it to a Verifier as a proof of validity of a corresponding Digital Credential.
```
```suggestion
    - The **Status Assertion flows**: these flows define the technical protocols for requesting and obtaining a Status Assertion and how the Wallet Instance SHOULD provide it to a Verifier as a proof of validity of a corresponding Digital Credential.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -113,7 +115,7 @@ Revocation Flows
 
 Depending on the different scenarios that may involve the revocation of a Digital Credential, different processes and technical flows may be implemented, according to national laws or Regulations of specific domains.
-The subsequent sections define the protocol interface between the Wallet Instances and the Issuers during the revocation request. The communication between the Issuers and other Entities is out-of-scope of this technical implementation profile.
+The subsequent sections define the protocol interface between the Wallet Instances and the Credential Issuers during the revocation request. The communication between the Credential Issuers and other Entities is out-of-scope of this technical implementation profile.
```
```suggestion
The subsequent sections define the protocol interface between the Wallet Instances and the Credential Issuers during the revocation request.
```

not clear which would be other Entities. Do you mean the authentic sources?

m-basili replied:
The revocation flow could be triggered by:

1. Legal Authority or law enforcement
2. Wallet Provider (if it enables this feature)
3. Authentic Source

The current version specify only the protocol interface between the Wallet Instance and the Credential Issuer. We probably should define protocol interfaces for the above entities in a future milestone.

@peppelinux and @fmarino-ipzs, do you agree?

SaraConsoliACN replied:
The mentioned sentence was already present in the current version of this specification: "The communication between the Issuers and other Entities is out-of-scope of this technical implementation profile." It was not introduced by my pull request, but if you prefer to remove it, I'll accept your review. WDYT?

peppelinux replied:
@m-basili issues created for future milestone 
https://github.com/italia/eudi-wallet-it-docs/issues/369

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -131,26 +133,30 @@ A Wallet Instance MUST request the revocation of a Digital Credential as defined
     Wallet Instance Initiated Revocation Flow
 
-**Step 1 (Credential Revocation Request)**: The Wallet Instance initiates the process by creating a Credential Revocation Request. This request includes a Digital Credential Proof of Possession as a JWT. It MUST be signed with the private key related to the public key contained within the Credential (such as the Issuer Signed JWT in the case of SD-JWT, or the MSO in the case of Mdoc CBOR). Then,  ...
+**Step 1 (Credential Revocation Request)**: The Wallet Instance initiates the process by creating a Credential Revocation Request. This request can be sent to a single Credential Issuer, regarding multiple Digital Credentials, and MUST contain a JSON object with the member `revocation_requests`.
```
```suggestion
**Step 1 (Credential Revocation Request)**: The Wallet Instance initiates the process by creating a Credential Revocation Request. This request MUST be sent to the Credential Issuer who has previously issued that Credential. The Credential Revocation Request MUST contain a JSON object with the member `revocation_requests`.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -131,26 +133,30 @@ A Wallet Instance MUST request the revocation of a Digital Credential as defined
-**Step 1 (Credential Revocation Request)**: The Wallet Instance initiates the process by creating a Credential Revocation Request. This request includes a Digital Credential Proof of Possession as a JWT. It MUST be signed with the private key related to the public key contained within the Credential (such as the Issuer Signed JWT in the case of SD-JWT, or the MSO in the case of Mdoc CBOR). Then,  ...
+**Step 1 (Credential Revocation Request)**: The Wallet Instance initiates the process by creating a Credential Revocation Request. This request can be sent to a single Credential Issuer, regarding multiple Digital Credentials, and MUST contain a JSON object with the member `revocation_requests`.
+
+The `revocation_requests` MUST be set with an array of strings, where each string within the array represents a Credential Revocation Request object.
```
```suggestion
The `revocation_requests` MUST be set with an array of strings, where each string within the array represents a Credential Revocation Request object, enabling the Wallet Instance to request multiple Credential Revocation Requests to a single Credential Issuer.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -131,26 +133,30 @@ A Wallet Instance MUST request the revocation of a Digital Credential as defined
+
+The `revocation_requests` MUST be set with an array of strings, where each string within the array represents a Credential Revocation Request object.
+
+It MUST be signed with the private key related to the public key contained within the Credential (such as the Credential Issuer Signed JWT in the case of SD-JWT, or the MSO in the case of Mdoc CBOR). Then, the Wallet Instance sends the request to the Credential Issuer as in the following non-normative example representing a Revocation Assertion Request array.
```
```suggestion
The request MUST be signed with the private key related to the public key contained within the Credential (such as the Credential Issuer Signed JWT in the case of SD-JWT, or the MSO in the case of Mdoc CBOR). Then, the Wallet Instance sends the request to the Credential Issuer as in the following non-normative example representing a Revocation Assertion Request array.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -160,27 +166,33 @@ Below, is given a non-normative example of a Credential PoP with decoded JWT hea
+      "credential_hash": $Issuer-Signed-JWT-Hash,
       "credential_hash_alg": "sha-256",
+      "credential_type": "PersonalIdentiticationData",
+      "user_identifier": "001",
+      "document_id": "PID-98765"
```
```suggestion
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -160,27 +166,33 @@ Below, is given a non-normative example of a Credential PoP with decoded JWT hea
-      "credential_hash": $Issuer-Signed-JWT-Hash
+      "credential_hash": $Issuer-Signed-JWT-Hash,
       "credential_hash_alg": "sha-256",
+      "credential_type": "PersonalIdentiticationData",
```
```suggestion
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -160,27 +166,33 @@ Below, is given a non-normative example of a Credential PoP with decoded JWT hea
     }
 
-**Step 2 (PoP verification)**: The Issuer verifies the signature of the PoP JWTs using the public key that was attested in the issued Digital Credential. If the verification is successful, it means that the Wallet Instance owns the private keys associated with the Digital Credential, and therefore is entitled to request its revocation.
+**Step 2 (PoP verification)**: The Credential Issuer verifies the signature of the PoP using the the confirmation method that was attested in the issued Digital Credential. If the verification is successful, it means that the Wallet Instance owns the private keys associated with the Digital Credential, and therefore is entitled to request its revocation.
```
```suggestion
**Step 2 (PoP verification)**: The Credential Issuer verifies the proof of possession of the Credential requested to be revoked, using the the confirmation method that was attested in the Credential. If the verification is successful the revocation request is allowed.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -160,27 +166,33 @@ Below, is given a non-normative example of a Credential PoP with decoded JWT hea
+**Step 2 (PoP verification)**: The Credential Issuer verifies the signature of the PoP using the the confirmation method that was attested in the issued Digital Credential. If the verification is successful, it means that the Wallet Instance owns the private keys associated with the Digital Credential, and therefore is entitled to request its revocation.
 
-**Step 3 (Credential Revocation)**: The Issuer revokes the Credential provided in the Credential PoP JWT. After the revocation, the Issuer MAY also send a notification to the User (e.g. using a User's email address, telephone number, or any other verified and secure communication channel), with all needed information related to the Credential revocation status update. This communication is out of ...
+**Step 3 (Credential Revocation)**: The Credential Issuer revokes the Credential provided in the revocation_requests object. After the revocation, the Credential Issuer MAY also send a notification to the User (e.g. using a User's email address, telephone number, or any other verified and secure communication channel), with all needed information related to the Credential revocation status update ...
```
```suggestion
**Step 3 (Credential Revocation)**: The Credential Issuer revokes the Credential provided in the Revocation Request object. After the revocation, the Credential Issuer MAY also send a notification to the User (e.g. using a User's email address, telephone number, or any other verified and secure communication channel), with all needed information related to the Credential revocation status update. This communication is out of scope of the current technical implementation profile. 
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -160,27 +166,33 @@ Below, is given a non-normative example of a Credential PoP with decoded JWT hea
+			"revocation_assertion_responses": ["${base64url(json({typ: revocation_assertion+jwt, ...}))}.payload.signature", ... ]
+		}
+ 
 Credential Revocation HTTP Request
```
is this duplicated with the previous one?

peppelinux replied:
nope.

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -503,56 +625,160 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
     * - **credential_hash_alg**
       - It MUST contain the Algorithm used for hashing the Digital Credential. The value SHOULD be set to `S256`.
       - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
+    * - **credential_type**
+      - Type of credential (e.g., Personal Identification Data, European Disability Card, European Health Insurance Card, Driver's License, etc.)
+      -
+    * - **user_identifier**
+      - 	Identifier used to avoid direct linking between the credential hash and the owner.
+      -
+    * - **document_id**
+      - 	Unique identifier of the document to which the attestation refers.
+      -
```
no

non deve contenere alcuna informazione relativa alla credenziale se non la sola prova di possesso della credenziale a cui è riferita?

m-basili replied:
```suggestion
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -503,56 +625,160 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+    * - **revoked**
+      - Boolean value indicating the status of Credential revocation. It MUST be set with the value `true` in revocation scenario.
+      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
+    * - **suspended**
+      -  Boolean value indicating the status of Credential suspension. It MUST be set with the value `false` in revocation scenario.
+      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
```
ad oggi non supportiamo la sospensione ...

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -503,56 +625,160 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
     - `[OIDC4VCI. Draft 13] <https://openid.bitbucket.io/connect/openid-4-verifiable-credential-issuance-1_0.html>`_, [:rfc:`7515`], [:rfc:`7517`].
   * -  **typ** 
-    - It MUST be set to `status-attestation+jwt`.
+    - It MUST be set to `status-assertion-request+jwt` when JWT format is used. It MUST be set to `status-assertion-request+cwt` when CWT format is used.
```
ad oggi non supportiamo CWT

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -190,18 +202,25 @@ The requests to the *Issuer Revocation endpoint* MUST be HTTP with method POST,
-      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_
+    * - **revocation_requests**
+      - It MUST be an array of strings, where each represents a Revocation Assertion Request object. Each element MUST contain a signed JWT/CWT as a cryptographic proof of possession to which the Digital Credential to be revoked shall be bound. See Section :ref:`Credential Proof of Possession <sec_revocation_credential_pop>` for more details.
+      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-assertions/01/>`_
```
```suggestion
      - `[OAuth Status Assertions] <https://datatracker.ietf.org/doc/draft-demarco-oauth-status-assertions>`_
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -212,142 +231,220 @@ The following HTTP Status Codes MUST be supported:
+      - **Reference**
+    * - **revocation_assertion_responses**
+      - the Revocation Assertions and or the Revocation Assertion Errors related to the request made by the Wallet Instance. 
+      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
```
```suggestion
      - `[OAuth Status Attestation] <https://datatracker.ietf.org/doc/draft-demarco-oauth-status-assertions>`_.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -212,142 +231,220 @@ The following HTTP Status Codes MUST be supported:
+      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
 
-  - *error*. The error code, as registerd in the table below.
+The Revocation Assertion object MUST contain the parameter ``credential_status_validity`` with the value set to ``false`` and the parameter ``revoked`` set to ``true``.
```
according to the status assertions only `credential_status_validity` is required, therefore, please remove all the other requirements ...

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -212,142 +231,220 @@ The following HTTP Status Codes MUST be supported:
+    "credential_hash": $CREDENTIAL-HASH,
+    "credential_hash_alg": "sha-256",
+    "error": "unsupported_hash_alg",
+    "error_description": "The hash algorithm set is not supported"
```
```suggestion
    "error_description": "The hash algorithm is not supported"
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -212,142 +231,220 @@ The following HTTP Status Codes MUST be supported:
+		"status_assertion_responses": ["${base64url(json({typ: status-assertion+jwt, ...}))}.payload.signature", ... ]
+	}
+
+The member `status_assertion_responses` MUST be an array of strings, where each of them represent a Status Assertion Response object as defined in `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
```
```suggestion
The member `status_assertion_responses` MUST be an array of strings, where each of them represent a Status Assertion Response object as defined in `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-oauth-status-assertions>`_.
```


instead of duplicating the URL as a constant, it is required to create a ref to a defined resource, as made with all tyhe other specs

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -456,43 +578,43 @@ Law-Enforcement Authorities or Third Parties authorized by national law, MAY req
       - **Reference**
     * - **typ**
-      - In case of revocation request it MUST be set to ``revocation-request+jwt``. In case of Status Attestation request it MUST be set to ``status-attestation-request+jwt``, according to `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
+      - In case of revocation request it MUST be set to ``revocation-request+{jwt,cwt}``. In case of Status Assertion request it MUST be set to ``status-assertion-request+{jwt,cwt}``, according to `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-assertions/01/>`_.
```
```suggestion
      - In case of revocation request it MUST be set to ``revocation-request+{jwt,cwt}``. In case of Status Assertion request it MUST be set to ``status-assertion-request+{jwt,cwt}``, according to `[OAuth Status Assertions] <https://datatracker.ietf.org/doc/draft-demarco-oauth-status-assertions>`_.
```

see my previous comment

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -456,43 +578,43 @@ Law-Enforcement Authorities or Third Parties authorized by national law, MAY req
       - :rfc:`7516#section-4.1.1`.
     * - **kid**
-      -  Unique identifier of the ``jwk`` inside the ``cnf`` claim of the Credential to be revoked, as base64url-encoded JWK Thumbprint value.
+      -  Unique identifier of the ``jwk`` or ``COSE_Key`` inside the ``cnf`` claim of the Credential to be revoked, as base64url-encoded JWK Thumbprint value, according to `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-assertions/01/>`_.
```
```suggestion
      -  Unique identifier of the ``jwk`` or ``COSE_Key`` inside the ``cnf`` claim of the Credential to be revoked, as base64url-encoded JWK Thumbprint value, according to `[OAuth Status Assertions] <https://datatracker.ietf.org/doc/draft-demarco-oauth-status-assertions>`_.
```

SaraConsoliACN committed:
Merge branch 'italia:versione-corrente' into SaraConsoliACN-statusAssertion

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
review suggestion applied

The suggestion mentioned in the comment https://github.com/italia/eudi-wallet-it-docs/pull/308#discussion_r1673884846 has been applied.

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

m-basili committed:
Update common_definitions.rst

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -11,26 +11,26 @@ For these reasons a robust mechanism for managing the life-cycle and the revocat
 Furthermore, it provides the technical details that the Verifiers MUST implement to verify, in a secure and reliable manner, the validity of a Digital Credential during the presentation phase.
 
-The verification of the validity of a Digital Credential is based on the `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_ specification. 
+The verification of the validity of a Digital Credential is based on the `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-oauth-status-assertions>`_ specification.
```
```suggestion
The verification of the validity of a Digital Credential is based on the `OAUTH-STATUS-ASSERTION`_ draft 02.
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -40,18 +40,20 @@ Operational Requirements
 -----------------------
 
-**The Status Attestation MUST:**
+**The Status Assertion:**
```
```suggestion
In addition to the requirements in Section 5 of `OAUTH-STATUS-ASSERTION`_ draft 02, **The Status Assertion:**
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -97,14 +99,14 @@ Credential Revocation Flows can start under different scenarios, such as:
-    - The Authentic Sources that for any administrative reasons update one or more attributes of a User, shall inform the Issuer of related Credentials. 
-    - The Issuers, for technical security reasons (e.g. in the case of compromised cryptographic keys, death of the User, etc.), can decide to revoke the Credentials.
+    - The Authentic Sources that for any update of one or more User attributes, shall inform the Credential Issuer that has previously requested those data for the issuance of a Credential about that User. 
+    - The Credential Issuers, for technical security reasons (e.g. in the case of compromised cryptographic keys, loss of device, etc.), SHOULD decide to revoke the Credentials.
```
```suggestion
    - The Credential Issuers, for technical security reasons (e.g. in the case of compromised cryptographic keys), SHOULD decide to revoke the Credentials.
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -97,14 +99,14 @@ Credential Revocation Flows can start under different scenarios, such as:
     - The User notifies an Authentic Source that one or more attributes are changed (e.g. the current resident address): in this case the Credentials SHALL be revoked, as they are no longer valid due to the change in attributes. 
```
```suggestion
    - The User notifies an Authentic Source that one or more attributes are changed (e.g. the current resident address): in this case the Credentials MUST be revoked, as they are no longer valid due to the change in attributes. 
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -97,14 +99,14 @@ Credential Revocation Flows can start under different scenarios, such as:
     - The Law-Enforcing Authorities, for the fulfillment of their functions and any other judicial reasons, may request the Authentic Source to revoke entitlements, licenses, certificates, identification documents, etc., which in turn leads to the revocation of any linked Credentials.
-    - The Authentic Sources that for any administrative reasons update one or more attributes of a User, shall inform the Issuer of related Credentials. 
-    - The Issuers, for technical security reasons (e.g. in the case of compromised cryptographic keys, death of the User, etc.), can decide to revoke the Credentials.
+    - The Authentic Sources that for any update of one or more User attributes, shall inform the Credential Issuer that has previously requested those data for the issuance of a Credential about that User. 
```
```suggestion
    - The Authentic Sources that for any update of one or more User attributes, SHOULD inform the Credential Issuer that has previously requested those data for the issuance of a Credential about that User. 
```

SaraConsoliACN committed:
a

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -190,18 +202,25 @@ The requests to the *Issuer Revocation endpoint* MUST be HTTP with method POST,
-      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_
+    * - **revocation_requests**
+      - It MUST be an array of strings, where each represents a Revocation Assertion Request object. Each element MUST contain a signed JWT/CWT as a cryptographic proof of possession to which the Digital Credential to be revoked shall be bound. See Section :ref:`Credential Proof of Possession <sec_revocation_credential_pop>` for more details.
+      - `[OAuth Status Assertions] <https://datatracker.ietf.org/doc/draft-demarco-oauth-status-assertions>`_
```
```suggestion
      - `OAUTH-STATUS-ASSERTION`_ draft 02.
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -190,18 +202,25 @@ The requests to the *Issuer Revocation endpoint* MUST be HTTP with method POST,
-      - It MUST contain a JWT proof of possession of the cryptographic key the Credential to be revoked shall be bound to. See Section :ref:`Credential Proof of Possession <sec_revocation_credential_pop>` for more details. 
-      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_
+    * - **revocation_requests**
+      - It MUST be an array of strings, where each represents a Revocation Assertion Request object. Each element MUST contain a signed JWT/CWT as a cryptographic proof of possession to which the Digital Credential to be revoked shall be bound. See Section :ref:`Credential Proof of Possession <sec_revocation_credential_pop>` for more details.
```
```suggestion
      - It MUST be an array of strings, where each represents a Revocation Assertion Request object. Each element MUST contain a signed JWT/CWT as a cryptographic proof of possession to which the Digital Credential to be revoked shall be bound. See Section :ref:`Credential Proof of Possession <sec_revocation_credential_pop>`_ for more details.
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -212,142 +231,220 @@ The following HTTP Status Codes MUST be supported:
     * - *400 Bad Request*
       - Error code and description
-      - The issuer cannot fulfill the request because of invalid parameters.
-    * - *404 Not Found*
```
Why HTTP status 404 Not Found was removed?

If the Credential doesn't exist we should rise an error to inform the wallet instance.
The Status Code 200 means that the credential is succesfully revoked by the Credential Issuer.

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -212,142 +231,220 @@ The following HTTP Status Codes MUST be supported:
+      - **Reference**
+    * - **revocation_assertion_responses**
+      - the Revocation Assertions and or the Revocation Assertion Errors related to the request made by the Wallet Instance. 
+      - `[OAuth Status Attestation] <https://datatracker.ietf.org/doc/draft-demarco-oauth-status-assertions>`_.
```
```suggestion
      - `OAUTH-STATUS-ASSERTION`_ draft 02`_.
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -212,142 +231,220 @@ The following HTTP Status Codes MUST be supported:
+      - the Revocation Assertions and or the Revocation Assertion Errors related to the request made by the Wallet Instance. 
+      - `[OAuth Status Attestation] <https://datatracker.ietf.org/doc/draft-demarco-oauth-status-assertions>`_.
+
+The Revocation Assertion object MUST contain the parameter ``credential_status_validity`` with the value set to ``false`` and the parameter ``revoked`` set to ``true``.
```
```suggestion
The Revocation Assertion object MUST contain the parameter ``credential_status_validity`` with the value set to ``false``.
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -212,142 +231,220 @@ The following HTTP Status Codes MUST be supported:
+    "credential_hash": $CREDENTIAL-HASH,
+    "credential_hash_alg": "sha-256",
+    "credential_status_validity": false,
+    "credential_status": {
+      "revoked": true,
+      "suspended": false
+    },
```
```suggestion
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -212,142 +231,220 @@ The following HTTP Status Codes MUST be supported:
+The `error` parameter for the Revocation Assertion Error object MUST be set with one of the values defined in the table below, in addition to the values specified in :rfc:`6749#section-5.2`:
 
-.. list-table:: 
+Here's the fixed table in reStructuredText (reST) format:
```
```suggestion
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -212,142 +231,220 @@ The following HTTP Status Codes MUST be supported:
-
-
-Below a non-normative example of an HTTP Response with an error.
+    * - ``credential_already_revoked``
+      - The Digital Credential is already revoked.
+    * - ``credential_updated``
+      - One or more information contained in the Digital Credential are changed. The `error_description` field SHOULD contain a human-readable text describing the general parameters updated without specifying each one.
+    * - ``credential_invalid``
+      - The Digital Credential is invalid. The `error_description` field SHOULD contain the reason of invalidation.
+    * - ``invalid_request_signature``
+      - The Revocation Assertion Request signature validation has failed. This error type is used when the proof of possession of the Digital Credential is found not valid within the Revocation Assertion Request.
+    * - ``credential_not_found``
+      - The `credential_hash` value provided in the Revocation Assertion Request doesn't match with any active Digital Credential.
+    * - ``unsupported_hash_alg``
+      - The hash algorithm set in `credential_hash_alg` is not supported.
+
```
It seems we already have a table with HTTP status code at line 225.
What is the meaning of these?

SaraConsoliACN replied:
> It seems we already have a table with HTTP status code at line 225. What is the meaning of these?

These status codes refer to the revocation assertion flow. The other table shows status assertion codes.

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -212,142 +231,220 @@ The following HTTP Status Codes MUST be supported:
+		"status_assertion_responses": ["${base64url(json({typ: status-assertion+jwt, ...}))}.payload.signature", ... ]
+	}
 
+The member `status_assertion_responses` MUST be an array of strings, where each of them represent a Status Assertion Response object as defined in `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
```
```suggestion
The member `status_assertion_responses` MUST be an array of strings, where each of them represent a Status Assertion Response object as defined in `OAUTH-STATUS-ASSERTION`_ draft 02`_.
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -356,32 +453,41 @@ The requests to the *Credential status endpoint* of the Issuers MUST be HTTP wit
-      - It MUST contain a signed JWT as a cryptographic proof of possession of the Digital Credential. See Section :ref:`Credential Proof of Possession <sec_revocation_credential_pop>` for more details. 
+    * - **status_assertion_requests**
+      - It MUST be an array of strings, where each of them represent a Status Assertion Request object. Each element MUST contain a signed JWT/CWT as a cryptographic proof of possession of the Digital Credential. See Section :ref:`Credential Proof of Possession <sec_revocation_credential_pop>` for more details. 
       - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_
```
```suggestion
      - `OAUTH-STATUS-ASSERTION`_ draft 02.
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -356,32 +453,41 @@ The requests to the *Credential status endpoint* of the Issuers MUST be HTTP wit
-    * - **credential_pop**
-      - It MUST contain a signed JWT as a cryptographic proof of possession of the Digital Credential. See Section :ref:`Credential Proof of Possession <sec_revocation_credential_pop>` for more details. 
+    * - **status_assertion_requests**
+      - It MUST be an array of strings, where each of them represent a Status Assertion Request object. Each element MUST contain a signed JWT/CWT as a cryptographic proof of possession of the Digital Credential. See Section :ref:`Credential Proof of Possession <sec_revocation_credential_pop>` for more details. 
```
```suggestion
      - It MUST be an array of strings, where each of them represent a Status Assertion Request object. Each element MUST contain a signed JWT/CWT as a cryptographic proof of possession of the Digital Credential. See Section :ref:`Credential Proof of Possession <sec_revocation_credential_pop>`_ for more details. 
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -356,32 +453,41 @@ The requests to the *Credential status endpoint* of the Issuers MUST be HTTP wit
-      - It MUST contain the Status Attestation as a signed JWT. 
+    * - **status_assertion_responses**
+      - the Status Assertions and or the Status Assertion Errors related to the request made by the Wallet Instance. 
       - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
```
```suggestion
      - `OAUTH-STATUS-ASSERTION`_ draft 02.
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -391,29 +497,29 @@ The following HTTP Status Codes MUST be supported:
     * - *404 Not Found*
       - 
-      - The Digital Credential can not be found by the Issuer.
+      - The Digital Credential may not be found by the Issuer, or the Status Attestation cannot be issued because the Credential has been revoked or updated.
```
Can we use the same approach for the credential revocation response?

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -456,43 +578,43 @@ Law-Enforcement Authorities or Third Parties authorized by national law, MAY req
       - **Reference**
     * - **typ**
-      - In case of revocation request it MUST be set to ``revocation-request+jwt``. In case of Status Attestation request it MUST be set to ``status-attestation-request+jwt``, according to `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
+      - In case of revocation request it MUST be set to ``revocation-request+{jwt,cwt}``. In case of Status Assertion request it MUST be set to ``status-assertion-request+{jwt,cwt}``, according to `[OAuth Status Assertions] <https://datatracker.ietf.org/doc/draft-demarco-oauth-status-assertions>`_.
```
```suggestion
      - In case of revocation request it MUST be set to ``revocation-request+{jwt,cwt}``. In case of Status Assertion request it MUST be set to ``status-assertion-request+{jwt,cwt}``, according to `OAUTH-STATUS-ASSERTION`_ draft 02.
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -456,43 +578,43 @@ Law-Enforcement Authorities or Third Parties authorized by national law, MAY req
       - :rfc:`7516#section-4.1.1`.
     * - **kid**
-      -  Unique identifier of the ``jwk`` inside the ``cnf`` claim of the Credential to be revoked, as base64url-encoded JWK Thumbprint value.
+      -  Unique identifier of the ``jwk`` or ``COSE_Key`` inside the ``cnf`` claim of the Credential to be revoked, as base64url-encoded JWK Thumbprint value, according to `[OAuth Status Assertions] <https://datatracker.ietf.org/doc/draft-demarco-oauth-status-assertions>`_.
```
```suggestion
      -  Unique identifier of the ``jwk`` or ``COSE_Key`` inside the ``cnf`` claim of the Credential to be revoked, as base64url-encoded JWK Thumbprint value, according to `OAUTH-STATUS-ASSERTION`_ draft 02.
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -503,56 +625,160 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+    - **Description**
+    - **Reference**
+  * - **alg**
+    - Algorithm used to verify the cryptographic signature of the Revocation Assertion. Revocation Assertion that do not need to be signed SHOULD set the `alg` value to `none` in according with `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
```
```suggestion
    - Algorithm used to verify the cryptographic signature of the Revocation Assertion. Revocation Assertion that do not need to be signed SHOULD set the `alg` value to `none` in according with `OAUTH-STATUS-ASSERTION`_ draft 02.
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -503,56 +625,160 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+    - `[OIDC4VCI. Draft 13] <https://openid.bitbucket.io/connect/openid-4-verifiable-credential-issuance-1_0.html>`_, [:rfc:`7515`], [:rfc:`7517`].
+  * -  **typ** 
+    - It MUST be set to `revocation-assertion-response+jwt` when JWT format is used. It MUST be set to `revocation-assertion-response+cwt` when CWT format is used.
+    - [:rfc:`7515`], [:rfc:`7517`], `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_..
```
```suggestion
    - [:rfc:`7515`], [:rfc:`7517`], `OAUTH-STATUS-ASSERTION`_ draft 02.
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -503,56 +625,160 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+    * - **credential_status_validity**
+      - Boolean value indicating the absolute validity of the Credential linked to the Status Assertion. It MUST be set with the value `false`.
+      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
+    * - **revoked**
+      - Boolean value indicating the status of Credential revocation. It MUST be set with the value `true` in revocation scenario.
+      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
+    * - **suspended**
+      -  Boolean value indicating the status of Credential suspension. It MUST be set with the value `false` in revocation scenario.
+      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
```
```suggestion
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -503,56 +625,160 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
     * - **credential_hash**
-      - Hash value of the Credential the Status Attestation is bound to.
+      - Hash value of the Credential the Status Assertion is bound to.
       - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
```
```suggestion
      - `OAUTH-STATUS-ASSERTION`_ draft 02.
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -503,56 +625,160 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+    * - **credential_status_validity**
+      - Boolean value indicating the absolute validity of the Credential linked to the Status Assertion. It is REQUIRED and it MUST be set with the value "false" or "true".
+      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
+    * - **revoked**
+      - Boolean value indicating the status of Credential revocation when the `credential_status_validity` is set to `false`. It MAY be set with the value `false` or `true`.
+      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
+    * - **suspended**
+      -  Boolean value indicating the status of Credential suspension. It MAY be set with the value `false` or `true`.
```
```suggestion
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -503,56 +625,160 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
     * - **credential_hash_alg**
-      - The Algorithm used for hashing the Credential to which the Status Attestation is bound. The value SHOULD be set to ``S256``.
+      - The Algorithm used for hashing the Credential to which the Status Assertion is bound. The value SHOULD be set to ``S256``.
+      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
```
```suggestion
      - `OAUTH-STATUS-ASSERTION`_ draft 02.
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -503,56 +625,160 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
+    * - **credential_status_validity**
+      - Boolean value indicating the absolute validity of the Credential linked to the Status Assertion. It is REQUIRED and it MUST be set with the value "false" or "true".
+      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
```
```suggestion
      - `OAUTH-STATUS-ASSERTION`_ draft 02.
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -503,56 +625,160 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
+    * - **suspended**
+      -  Boolean value indicating the status of Credential suspension. It MAY be set with the value `false` or `true`.
       - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
```
```suggestion
      - `OAUTH-STATUS-ASSERTION`_ draft 02.
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -503,56 +625,160 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+    - **Description**
+    - **Reference**
+  * - **alg**
+    - Algorithm used to verify the cryptographic signature of the Assertion Error. Assertion Error that do not need to be signed SHOULD set the `alg` value to `none` in according with `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
```
```suggestion
    - Algorithm used to verify the cryptographic signature of the Assertion Error. Assertion Error that do not need to be signed SHOULD set the `alg` value to `none` in according with `OAUTH-STATUS-ASSERTION`_ draft 02.
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -503,56 +625,160 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+    - `[OIDC4VCI. Draft 13] <https://openid.bitbucket.io/connect/openid-4-verifiable-credential-issuance-1_0.html>`_, [:rfc:`7515`], [:rfc:`7517`].
+  * -  **typ** 
+    - It MUST be set to `status-assertion-response+jwt` or `revocation-assertion-response+jwt` when JWT format is used. It MUST be set to `status-assertion-response+cwt` or `revocation-assertion-response+cwt` when CWT format is used.
+    - [:rfc:`7515`], [:rfc:`7517`], `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
```
```suggestion
    - [:rfc:`7515`], [:rfc:`7517`], `OAUTH-STATUS-ASSERTION`_ draft 02.
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -503,56 +625,160 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+      - Unique identifier for the JWT.
+      - `[RFC7519, Section 4.1.7] <https://www.iana.org/go/rfc7519>`_.
+    * - **error**
+      - Status code returned from the Credential Issuer after revocation. The value SHOULD be assigned with one of the error types defined in  {{RFC6749}}[Section 5.2]<https://tools.ietf.org/html/rfc6749#section-5.2> or defined in `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_ 
```
```suggestion
      - Status code returned from the Credential Issuer after revocation. The value SHOULD be assigned with one of the error types defined in  {{RFC6749}}[Section 5.2]<https://tools.ietf.org/html/rfc6749#section-5.2> or defined in `OAUTH-STATUS-ASSERTION`_ draft 02. 
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -503,56 +625,160 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+      - `[RFC7519, Section 4.1.7] <https://www.iana.org/go/rfc7519>`_.
+    * - **error**
+      - Status code returned from the Credential Issuer after revocation. The value SHOULD be assigned with one of the error types defined in  {{RFC6749}}[Section 5.2]<https://tools.ietf.org/html/rfc6749#section-5.2> or defined in `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_ 
+      - `[RFC6749, Section 5.2] <https://tools.ietf.org/html/rfc6749#section-5.2>`_, `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_
```
```suggestion
      - `[RFC6749, Section 5.2] <https://tools.ietf.org/html/rfc6749#section-5.2>`_, `OAUTH-STATUS-ASSERTION`_ draft 02
```

m-basili commented on docs/en/revocation-lists.rst:
```diff
@@ -503,56 +625,160 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+      - `[RFC6749, Section 5.2] <https://tools.ietf.org/html/rfc6749#section-5.2>`_, `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_
+    * - **error_description**
+      - Text that clarifies the nature of the error, such as attribute changes, revocation reasons, in relation to the `error` value.
+      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
```
```suggestion
      - `OAUTH-STATUS-ASSERTION`_ draft 02.
```

m-basili reviewed (changes requested):
I made some suggestions and I put some comments to be discussed. 
@SaraConsoliACN could you please remove CWT reference as we doesn't support this?

SaraConsoliACN committed:
Accepted review suggestion

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
external link fixed

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
external link fixed

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
external link fixed

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
external link fixed

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
external link fixed

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
external link fixed

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
external link fixed

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
external link fixed

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
external link fixed

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
external link fixed

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
external link fixed

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
external link fixed

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
external link fixed

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
external link fixed

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
Commit suggestion applied

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
external link fixed

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
external link fixed

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
external link fixed

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
external link fixed

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
external link fixed

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
external link fixed

Co-authored-by: m-basili <m.basili@ipzs.it>

SaraConsoliACN committed:
cwt occurences removed

As CWT is not supported, all occurrences of CWT have been removed as requested in the comment https://github.com/italia/eudi-wallet-it-docs/pull/308/files#r1674043524

m-basili committed:
fix: table error fixed

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -504,55 +612,138 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+      - `[RFC7519, Section 4.1.7] <https://www.iana.org/go/rfc7800>`_.
+    * - **credential_status_validity**
+      - Boolean value indicating the absolute validity of the Credential linked to the Status Assertion. It MUST be set with the value `false`.
+      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
```
```suggestion
      - `OAUTH-STATUS-ASSERTION`_.
```

please use a reference not a static url

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -504,55 +612,138 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
-      - Hash value of the Credential the Status Attestation is bound to.
-      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
+      - Hash value of the Credential the Status Assertion is bound to.
+      - `OAUTH-STATUS-ASSERTION`_ draft 02.
```
```suggestion
      - `OAUTH-STATUS-ASSERTION`_.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -504,55 +612,138 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
-      - The Algorithm used for hashing the Credential to which the Status Attestation is bound. The value SHOULD be set to ``S256``.
-      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
+      - The Algorithm used for hashing the Credential to which the Status Assertion is bound. The value SHOULD be set to ``S256``.
+      - `OAUTH-STATUS-ASSERTION`_ draft 02.
```
```suggestion
      - `OAUTH-STATUS-ASSERTION`_.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -504,55 +612,138 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+      - `OAUTH-STATUS-ASSERTION`_ draft 02.
+    * - **credential_status_validity**
+      - Boolean value indicating the absolute validity of the Credential linked to the Status Assertion. It is REQUIRED and it MUST be set with the value "false" or "true".
+      - `OAUTH-STATUS-ASSERTION`_ draft 02.
```
```suggestion
      - `OAUTH-STATUS-ASSERTION`_.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -504,55 +612,138 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+    - **Description**
+    - **Reference**
+  * - **alg**
+    - Algorithm used to verify the cryptographic signature of the Assertion Error. Assertion Error that do not need to be signed SHOULD set the `alg` value to `none` in according with `OAUTH-STATUS-ASSERTION`_ draft 02.
```
```suggestion
    - Algorithm used to verify the cryptographic signature of the Assertion Error. Assertion Error that do not need to be signed SHOULD set the `alg` value to `none` in according with `OAUTH-STATUS-ASSERTION`_.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -504,55 +612,138 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+    - `[OIDC4VCI. Draft 13] <https://openid.bitbucket.io/connect/openid-4-verifiable-credential-issuance-1_0.html>`_, [:rfc:`7515`], [:rfc:`7517`].
+  * -  **typ** 
+    - It MUST be set to `status-assertion-response+jwt` or `revocation-assertion-response+jwt` when JWT format is used.
+    - [:rfc:`7515`], [:rfc:`7517`], `OAUTH-STATUS-ASSERTION`_ draft 02.
```
```suggestion
    - [:rfc:`7515`], [:rfc:`7517`], `OAUTH-STATUS-ASSERTION`_ .
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -504,55 +612,138 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+      - Unique identifier for the JWT.
+      - `[RFC7519, Section 4.1.7] <https://www.iana.org/go/rfc7519>`_.
+    * - **error**
+      - Status code returned from the Credential Issuer after revocation. The value SHOULD be assigned with one of the error types defined in  {{RFC6749}}[Section 5.2]<https://tools.ietf.org/html/rfc6749#section-5.2> or defined in `OAUTH-STATUS-ASSERTION`_ draft 02. 
```
```suggestion
      - Status code returned from the Credential Issuer after revocation. The value SHOULD be assigned with one of the error types defined in  {{RFC6749}}[Section 5.2]<https://tools.ietf.org/html/rfc6749#section-5.2> or defined in `OAUTH-STATUS-ASSERTION`_. 
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -504,55 +612,138 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+      - `[RFC7519, Section 4.1.7] <https://www.iana.org/go/rfc7519>`_.
+    * - **error**
+      - Status code returned from the Credential Issuer after revocation. The value SHOULD be assigned with one of the error types defined in  {{RFC6749}}[Section 5.2]<https://tools.ietf.org/html/rfc6749#section-5.2> or defined in `OAUTH-STATUS-ASSERTION`_ draft 02. 
+      - `[RFC6749, Section 5.2] <https://tools.ietf.org/html/rfc6749#section-5.2>`_, `OAUTH-STATUS-ASSERTION`_ draft 02
```
```suggestion
      - `[RFC6749, Section 5.2] <https://tools.ietf.org/html/rfc6749#section-5.2>`_, `OAUTH-STATUS-ASSERTION`_
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -504,55 +612,138 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+      - `[RFC6749, Section 5.2] <https://tools.ietf.org/html/rfc6749#section-5.2>`_, `OAUTH-STATUS-ASSERTION`_ draft 02
+    * - **error_description**
+      - Text that clarifies the nature of the error, such as attribute changes, revocation reasons, in relation to the `error` value.
+      - `OAUTH-STATUS-ASSERTION`_ draft 02.
```
```suggestion
      - `OAUTH-STATUS-ASSERTION`_.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -504,55 +612,138 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+    - `[OIDC4VCI. Draft 13] <https://openid.bitbucket.io/connect/openid-4-verifiable-credential-issuance-1_0.html>`_, [:rfc:`7515`], [:rfc:`7517`].
+  * -  **typ** 
+    - It MUST be set to `revocation-assertion-response+jwt` when JWT format is used.
+    - [:rfc:`7515`], [:rfc:`7517`], `OAUTH-STATUS-ASSERTION`_ draft 02.
```
```suggestion
    - [:rfc:`7515`], [:rfc:`7517`], `OAUTH-STATUS-ASSERTION`_.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -504,55 +612,138 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
+    - **Description**
+    - **Reference**
+  * - **alg**
+    - Algorithm used to verify the cryptographic signature of the Revocation Assertion. Revocation Assertion that do not need to be signed SHOULD set the `alg` value to `none` in according with `OAUTH-STATUS-ASSERTION`_ draft 02.
```
```suggestion
    - Algorithm used to verify the cryptographic signature of the Revocation Assertion. Revocation Assertion that do not need to be signed SHOULD set the `alg` value to `none` in according with `OAUTH-STATUS-ASSERTION`_.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -504,55 +612,138 @@ The Credential Proof of Possession (**credential_pop**) MUST be a JWT that MUST
       - It MUST contain the Algorithm used for hashing the Digital Credential. The value SHOULD be set to `S256`.
       - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
```
```suggestion
      - `OAUTH-STATUS-ASSERTION`_.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -456,40 +564,40 @@ Law-Enforcement Authorities or Third Parties authorized by national law, MAY req
       - :rfc:`7516#section-4.1.1`.
     * - **kid**
-      -  Unique identifier of the ``jwk`` inside the ``cnf`` claim of the Credential to be revoked, as base64url-encoded JWK Thumbprint value.
+      -  Unique identifier of the ``jwk`` or ``COSE_Key`` inside the ``cnf`` claim of the Credential to be revoked, as base64url-encoded JWK Thumbprint value, according to `OAUTH-STATUS-ASSERTION`_ draft 02.
```
```suggestion
      -  Unique identifier of the ``jwk`` or ``COSE_Key`` inside the ``cnf`` claim of the Credential to be revoked, as base64url-encoded JWK Thumbprint value, according to `OAUTH-STATUS-ASSERTION`_.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -456,40 +564,40 @@ Law-Enforcement Authorities or Third Parties authorized by national law, MAY req
       - **Reference**
     * - **typ**
-      - In case of revocation request it MUST be set to ``revocation-request+jwt``. In case of Status Attestation request it MUST be set to ``status-attestation-request+jwt``, according to `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
+      - In case of revocation request it MUST be set to ``revocation-request+jwt``. In case of Status Assertion request it MUST be set to ``status-assertion-request+jwt``, according to `OAUTH-STATUS-ASSERTION`_ draft 02.
```
```suggestion
      - In case of revocation request it MUST be set to ``revocation-request+jwt``. In case of Status Assertion request it MUST be set to ``status-assertion-request+jwt``, according to `OAUTH-STATUS-ASSERTION`_ .
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -356,31 +439,40 @@ The requests to the *Credential status endpoint* of the Issuers MUST be HTTP wit
-      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_.
+    * - **status_assertion_responses**
+      - the Status Assertions and or the Status Assertion Errors related to the request made by the Wallet Instance. 
+      - `OAUTH-STATUS-ASSERTION`_ draft 02.
```
```suggestion
      - `OAUTH-STATUS-ASSERTION`_.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -356,31 +439,40 @@ The requests to the *Credential status endpoint* of the Issuers MUST be HTTP wit
-      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_
+    * - **status_assertion_requests**
+      - It MUST be an array of strings, where each of them represent a Status Assertion Request object. Each element MUST contain a signed JWT as a cryptographic proof of possession of the Digital Credential. See Section :ref:`Credential Proof of Possession <sec_revocation_credential_pop>`_ for more details. 
+      - `OAUTH-STATUS-ASSERTION`_ draft 02.
```
```suggestion
      - `OAUTH-STATUS-ASSERTION`_ .
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -212,142 +222,215 @@ The following HTTP Status Codes MUST be supported:
+		"status_assertion_responses": ["${base64url(json({typ: status-assertion+jwt, ...}))}.payload.signature", ... ]
+	}
+
+The member `status_assertion_responses` MUST be an array of strings, where each of them represent a Status Assertion Response object as defined in `OAUTH-STATUS-ASSERTION`_ draft 02`_.
```
```suggestion
The member `status_assertion_responses` MUST be an array of strings, where each of them represent a Status Assertion Response object as defined in `OAUTH-STATUS-ASSERTION`_.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -190,18 +193,25 @@ The requests to the *Issuer Revocation endpoint* MUST be HTTP with method POST,
-      - `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_
+    * - **revocation_requests**
+      - It MUST be an array of strings, where each represents a Revocation Assertion Request object. Each element MUST contain a signed JWT as a cryptographic proof of possession to which the Digital Credential to be revoked shall be bound. See Section :ref:`Credential Proof of Possession <sec_revocation_credential_pop>`_ for more details.
+      - `OAUTH-STATUS-ASSERTION`_ draft 02.
```
```suggestion
      - `OAUTH-STATUS-ASSERTION`_ .
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -40,18 +40,14 @@ Operational Requirements
 -----------------------
 
-**The Status Attestation MUST:**
+In addition to the requirements in Section 5 of `OAUTH-STATUS-ASSERTION`_ draft 02, **The Status Assertion:**
```
```suggestion
In addition to the requirements in Section 5 of `OAUTH-STATUS-ASSERTION`_, **The Status Assertion:**
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -11,26 +11,26 @@ For these reasons a robust mechanism for managing the life-cycle and the revocat
 Furthermore, it provides the technical details that the Verifiers MUST implement to verify, in a secure and reliable manner, the validity of a Digital Credential during the presentation phase.
 
-The verification of the validity of a Digital Credential is based on the `[OAuth Status Attestation draft 01] <https://datatracker.ietf.org/doc/draft-demarco-status-attestations/01/>`_ specification. 
+The verification of the validity of a Digital Credential is based on the `OAUTH-STATUS-ASSERTION`_ draft 02.
```
```suggestion
The verification of the validity of a Digital Credential is based on the `OAUTH-STATUS-ASSERTION`_.
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -212,142 +222,215 @@ The following HTTP Status Codes MUST be supported:
+      - **Reference**
+    * - **revocation_assertion_responses**
+      - the Revocation Assertions and or the Revocation Assertion Errors related to the request made by the Wallet Instance. 
+      - `OAUTH-STATUS-ASSERTION`_ draft 02`_.
```
```suggestion
      - `OAUTH-STATUS-ASSERTION`_ .
```

SaraConsoliACN committed:
Applied code review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Applied code review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Applied code review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Applied code review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Applied code review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Applied code review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Applied code review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Applied code review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Applied code review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Applied code review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Applied code review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Applied code review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Applied code review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Applied code review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Applied code review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Applied code review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Applied code review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Applied code review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Applied code review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Applied code review suggestion

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

m-basili committed:
fix: Alignment with the current draft status

m-basili commented:
@peppelinux @SaraConsoliACN, I made last commits to align with the current draft of status assertion, as a note credential status validity is still in a PR and is not included in the draft 2 yet.

SaraConsoliACN committed:
external links fixed

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -11,26 +11,26 @@ For these reasons a robust mechanism for managing the life-cycle and the revocat
 - verification of the Digital Credential validity status in both online and offline scenarios;
 - privacy-preserving, according to the following evidences:
 
   1. the Verifier cannot check over time the validity of a given  Digital Credential related to the User;
```
```suggestion
  1. the Verifier can check the validity of the Credential during the presentation phase. It is not able to check the validity of a given Digital Credential related to the User over time and out of the scope of the User authentication;
```

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -11,26 +11,26 @@ For these reasons a robust mechanism for managing the life-cycle and the revocat
 
   1. the Verifier cannot check over time the validity of a given  Digital Credential related to the User;
-  2. the Issuers cannot track when and where a Digital Credential is verified;
+  2. the Credential Issuers cannot track when and where a Digital Credential is verified;
```
```suggestion
  2. the Credential Issuers is not able to know to which Verifier the Digital Credential or the Status Assertion will be presented;
```

SaraConsoliACN committed:
Apply suggestions from code review

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Apply suggestions from code review

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Duplicated RFC 7519 removed

RFC 7519 reference retained only in the standards.rst file.

SaraConsoliACN committed:
RFC 7519 referencing fixed

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -160,27 +160,30 @@ Below, is given a non-normative example of a Credential PoP with decoded JWT hea
       "jti": "6f204f7e-e453-4dfd-814e-9d155319408c",
-      "credential_hash": $Issuer-Signed-JWT-Hash
+      "credential_hash": $Issuer-Signed-JWT-Hash,
       "credential_hash_alg": "sha-256",
```
```suggestion
      "credential_hash_alg": "sha-256"
```

SaraConsoliACN committed:
Apply suggestions from code review

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN committed:
Merge branch 'versione-corrente' into SaraConsoliACN-statusAssertion

SaraConsoliACN committed:
Merge branch 'versione-corrente' into SaraConsoliACN-statusAssertion

SaraConsoliACN committed:
Merge branch 'versione-corrente' into SaraConsoliACN-statusAssertion

peppelinux commented on docs/en/revocation-lists.rst:
```diff
@@ -212,142 +216,222 @@ The following HTTP Status Codes MUST be supported:
+    "credential_status_validity": false,
+    "cnf": {
+      "jwk": {
+        "alg": "ES256",
```
```suggestion
```

SaraConsoliACN committed:
Apply suggestions from code review

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

SaraConsoliACN commented:
fixes #265
